test/e2e/federated-service.go

```diff
@@ -192,6 +192,11 @@ var _ = framework.KubeDescribe("Service [Feature:Federation]", func() {
 // cluster. If the shard count, i.e. numSvcs is expected to be at least one, then
 // it also checks if the first shard's name and spec matches that of the given service.
 func waitForFederatedServiceShard(cs *release_1_3.Clientset, namespace string, service *api.Service, numSvcs int) {
+	v1service := &v1.Service{}
+	if err := api.Scheme.Convert(service, v1service); err != nil {
```

Comment:
sigh this is not good.
This is because we are using unversioned clientset at one place (createService gets unversioned) and versioned one at another (cs here is versioned).
The right fix is to use versioned clientset everywhere.